vcd/resource_vcd_nsxt_app_port_profile.go

```diff
@@ -0,0 +1,394 @@
+package vcd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var appPortDefinition = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"protocol": {
+			Required:     true,
+			Type:         schema.TypeString,
+			ValidateFunc: validation.StringInSlice([]string{"ICMPv4", "ICMPv6", "TCP", "UDP"}, false),
+		},
+		"port": {
+			Optional:    true,
+			Type:        schema.TypeSet,
+			Description: "Set of ports or ranges",
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+	},
+}
+
+func resourceVcdNsxtAppPortProfile() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdNsxtAppPortProfileCreate,
+		ReadContext:   resourceVcdNsxtAppPortProfileRead,
+		UpdateContext: resourceVcdNsxtAppPortProfileUpdate,
+		DeleteContext: resourceVcdNsxtAppPortProfileDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNsxtAppPortProfileImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Application Port Profile name",
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Application Port Profile description",
+			},
+			"scope": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "Scope - 'PROVIDER' or 'TENANT'",
```

Comment:
Just in case: 'SYSTEM' not applicable here?

Reply:
Yes, because SYSTEM Application Port Profiles are "read-only". We cannot "create" a resource, just read it (that is why data source supports it)